src/test/java/org/broadinstitute/hellbender/tools/walkers/sv/SVConcordanceIntegrationTest.java

```diff
@@ -227,4 +227,152 @@ public void testSelf() {
+        final ClusteringParameters depthParameters = ClusteringParameters.createDepthParameters(0.5, 2000, 0);
+        final ClusteringParameters mixedParameters = ClusteringParameters.createMixedParameters(0.1, 2000, 0);
+        final ClusteringParameters pesrParameters = ClusteringParameters.createPesrParameters(0.1, 500, 0);
+        final SVConcordanceLinkage linkage = new SVConcordanceLinkage(dictionary);
+        linkage.setDepthOnlyParams(depthParameters);
+        linkage.setMixedParams(mixedParameters);
+        linkage.setEvidenceParams(pesrParameters);
+
+        final Comparator<VariantContext> idComparator = Comparator.comparing(VariantContext::getID);
+
+        final List<VariantContext> inputVariants = VariantContextTestUtils.readEntireVCFIntoMemory(evalVcfPath).getValue()
+                .stream()
+                .sorted(idComparator)
+                .collect(Collectors.toList());
+        final List<VariantContext> outputVariants = outputVcf.getValue().stream().sorted(idComparator).collect(Collectors.toList());
+
+        Assert.assertEquals(outputVariants.size(), inputVariants.size());
+        for (int i = 0; i < outputVariants.size(); i++) {
+            final VariantContext outputVariant = outputVariants.get(i);
+            final VariantContext expectedVariant = inputVariants.get(i);
+            Assert.assertEquals(outputVariant.getID(), expectedVariant.getID());
+            Assert.assertEquals(outputVariant.getContig(), expectedVariant.getContig());
+            Assert.assertEquals(outputVariant.getStart(), expectedVariant.getStart());
+            Assert.assertEquals(outputVariant.getEnd(), expectedVariant.getEnd());
+            Assert.assertEquals(outputVariant.getAlleles(), expectedVariant.getAlleles());
+            final String svtype = outputVariant.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "test_default");
+            Assert.assertEquals(svtype, expectedVariant.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "expected_default"));
+            // check the variant matched itself with perfect concordance
+            if (outputVariant.getAttributeAsString(Concordance.TRUTH_STATUS_VCF_ATTRIBUTE, "test_default") == ConcordanceState.TRUE_POSITIVE.getAbbreviation()) {
+                Assert.assertEquals(outputVariant.getAttributeAsString(GATKSVVCFConstants.TRUTH_VARIANT_ID_INFO, ""), outputVariant.getID());
+                if (svtype.equals(GATKSVVCFConstants.StructuralVariantAnnotationType.CNV.toString())) {
+                    MathUtilsUnitTest.assertEqualsDoubleSmart(outputVariant.getAttributeAsDouble(GATKSVVCFConstants.COPY_NUMBER_CONCORDANCE_INFO, -1.), 1.0, TOLERANCE);
+                } else {
+                    MathUtilsUnitTest.assertEqualsDoubleSmart(outputVariant.getAttributeAsDouble(GATKSVVCFConstants.GENOTYPE_CONCORDANCE_INFO, -1.), 1.0, TOLERANCE);
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testSelfEvalSubset() {
```

Comment:
There's a lot of duplicate code here, so I might be inclined to suggest a data provider, but it's also sort of a weird case where we would have a genericified test that expects perfect concordance.

Reply:
Okay I've created a `assertPerfectConcordance()` to do the checking at least.